ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -316,11 +319,12 @@ def record_environment_stats(
         :param worker_id:
         :return:
         """
-        for stat_name, (val, agg_type) in env_stats.items():
-            if agg_type == StatsAggregationMethod.AVERAGE:
-                self.stats_reporter.add_stat(stat_name, val)
-            elif agg_type == StatsAggregationMethod.MOST_RECENT:
-                # In order to prevent conflicts between multiple environments,
-                # only stats from the first environment are recorded.
-                if worker_id == 0:
-                    self.stats_reporter.set_stat(stat_name, val)
+        for stat_name, value_list in env_stats.items():
+            for val, agg_type in value_list:
+                if agg_type == StatsAggregationMethod.AVERAGE:
+                    self.stats_reporter.add_stat(stat_name, val)
+                elif agg_type == StatsAggregationMethod.MOST_RECENT:
+                    # In order to prevent conflicts between multiple environments,
+                    # only stats from the first environment are recorded.
+                    if worker_id == 0:
+                        self.stats_reporter.set_stat(stat_name, val)
```

Comment:
Kind of weird to set_stats for each element in value_list since we will only keep the last one. I would do the for loop `for val, agg_type in value_list:` inside the `if agg_type == StatsAggregationMethod.AVERAGE:` and only set_stat for the last element in the case `elif agg_type == StatsAggregationMethod.MOST_RECENT:`

Reply:
It's possible (but not recommended) to mix MOST_RECENT and AVERAGE for the same key. So if you passed
```
[
  (1, AVERAGE),
  (2, MOST_RECENT),
  (3, AVERAGE)
]
```
I think this is the most sensible way to handle it. You'd end up appending 1, then dropping that and replacing with 2, and finally appending 3.